File: captainhcg/go/80-remove-duplicates-from-sorted-array-ii.go
func removeDuplicates(nums []int) int {
    lastIdx, preVal, cnt := 0, 0, 0
    for idx := 0; idx < len(nums); idx += 1{
        v := nums[idx]
        if v != preVal{
            cnt = 1
            preVal = v
        } else {
            cnt += 1
        }
        if cnt <= 2{
            nums[lastIdx] = v
            lastIdx += 1
        }
    }
    return lastIdx
}

File: bwu/array/80-remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        if n < 3:
            return n
            
        idx, curr = 2, 2
        while curr < n:
            if nums[curr] != nums[idx - 2]:
                nums[idx] = nums[curr]
                idx += 1
            
            curr += 1
            
        return idx
